_episodes/07-find.md

```diff
@@ -382,6 +382,65 @@ Miscellaneous:
 > {: .solution}
 {: .challenge}
 
+> ## Using grep in a loop
+> Navigate to Nelle's `data-shell/data/pdb/` directory, which contains Protein Data Bank files.
+> Write a `for` loop which uses `echo` and `grep` to print each file name followed by the number
+> of atoms in the compound.
+>
+> Here's the beginning and end of the output your loop should return:
+> ```
+> aldrin.pdb 26
+> ammonia.pdb 4
+> ascorbic-acid.pdb 20
+> ...
+> vanillin.pdb 19
+> vinyl-chloride.pdb 6
+> vitamin-a.pdb 51
+> ```
+> {: .output}
+>
+> Hint: Read `man echo` for an option to suppress newlines
+> and `man grep` for an option to produce counts only.
+>
+> > ## Solution
+> > ```
+> > for file in *.pdb
+> >	do
+> >		echo -n $file " "
+> >		grep -c ATOM $file
+> >	done
+> > ```
+> > {: .language-bash}
+> {: .solution}
+>
+> Extend the previous solution to sort the output by the number of atoms.
+> Here's the beginning and end of the output:
+>
+> ~~~
+> ammonia.pdb 4
+> methane.pdb 5
+> methanol.pdb 6
+> ...
+> cholesterol.pdb 74
+> heme.pdb 75
+> lanoxin.pdb 121
+> ~~~
+> {: .output}
+>
+> Hint: Read `man sort` and look at the `-k` option to sort by a field.
```

Comment:
I think this is going to be difficult. The `man` entry for `-k` just says...

```
 -k, --key=KEYDEF
              sort via a key; KEYDEF gives location and type
```

If they search the man page for `KEYDEF`, they'll find...

```
   KEYDEF is F[.C][OPTS][,F[.C][OPTS]] for start and stop position, where F is a field number  and  C  a  character
   position  in  the field; both are origin 1, and the stop position defaults to the line's end.  If neither -t nor
   -b is in effect, characters in a field are counted from the beginning of the preceding whitespace.  OPTS is  one
   or  more  single-letter ordering options [bdfgiMhnRrV], which override global ordering options for that key.  If
   no key is given, use the entire line as the key.  Use --debug to diagnose incorrect key usage.
```
If I didn't already know what `-k` did, I would have no idea what any of this meant.